File: src/chrome/browser/resources/settings/chromeos/metrics_recorder.js
// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * @fileoverview
 * Provides functions used for recording user actions within settings.
 * Also provides a way to inject a test implementation for verifying
 * user action recording.
 */
cr.define('settings', function() {
  /** @type {?chromeos.settings.mojom.UserActionRecorderInterface} */
  let userActionRecorder = null;

  /**
   * @param {!chromeos.settings.mojom.UserActionRecorderInterface}
   *     testRecorder
   */
  function setUserActionRecorderForTesting(testRecorder) {
    userActionRecorder = testRecorder;
  }

  /**
   * @return {!chromeos.settings.mojom.UserActionRecorderInterface}
   */
  function getRecorder() {
    if (userActionRecorder) {
      return userActionRecorder;
    }

    userActionRecorder = chromeos.settings.mojom.UserActionRecorder.getRemote();
    return userActionRecorder;
  }

  function recordPageFocus() {
    getRecorder().recordPageFocus();
  }

  function recordPageBlur() {
    getRecorder().recordPageBlur();
  }

  function recordClick() {
    getRecorder().recordClick();
  }

  function recordNavigation() {
    getRecorder().recordNavigation();
  }

  function recordSearch() {
    getRecorder().recordSearch();
  }

  function recordSettingChange() {
    getRecorder().recordSettingChange();
  }

  // #cr_define_end
  return {
    setUserActionRecorderForTesting,
    recordPageFocus,
    recordPageBlur,
    recordClick,
    recordNavigation,
    recordSearch,
    recordSettingChange,
  };
});

File: src/chrome/browser/share/android/java/src/org/chromium/chrome/browser/share/send_tab_to_self/MetricsRecorder.java
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.share.send_tab_to_self;

import org.chromium.base.annotations.JNINamespace;
import org.chromium.base.annotations.NativeMethods;

/**
 * Class that captures all the metrics needed for Send Tab To Self on Android.
 */
@JNINamespace("send_tab_to_self")
class MetricsRecorder {
    public static void recordDeviceClickedInShareSheet() {
        MetricsRecorderJni.get().recordDeviceClickedInShareSheet();
    }

    @NativeMethods
    interface Natives {
        void recordDeviceClickedInShareSheet();
    }
}
